### PR TITLE
implemented video streaming support

### PR DESCRIPTION
**Pull request by miguelgarcia**
### Description

* Implemented video streaming capabilities into the Edge SDK using OpenCV, but without making it a hard dependency.
* Included example
* There are some improvements pending, like honoring the settings from `set_state`, but at least the implemented solution is "smart" enough to stream video only when requested from the platform side, minimizing bandwidth usage.

https://user-images.githubusercontent.com/398428/213211277-4c4da568-dcf0-4b62-a8e5-79c3238505bc.mp4

### CODE REVIEW

**Comment by miguelgarcia** [comment]
Hey @leandropineda can you take a look at this when you have a chance ? Thanks a lot!

File: inorbit_edge/video.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    logging.getLogger(__name__).warning(
+        "Could not import cv2. Video support won't work"
+    )
+
+
+class Camera:
+    """Interface that all camera classes must implement"""
+
+    def __init__(self):
+        raise Exception("Abstract class")
```
Comment (leandropineda) [inline inorbit_edge/video.py:30]: I'd use the ABC module for abstract classes https://docs.python.org/3/library/abc.html

```python
from abc import ABC, abstractmethod
 
class Camera(ABC):
    def __init__(self):
        super().__init__()
    
    @abstractmethod
    def open(self):
        pass

    ...
```

File: inorbit_edge/tests/demo/README.md [hunk 2]
```diff
@@ -4,14 +4,16 @@ Code for generating synthetic data for simulating a fleet of robots. It uses the
...
+Export required environment variables and execute the `example.py` script. Use the `virtualenv` used on the `CONTRIBUTING.md` guide.
 
 ```bash
 export INORBIT_URL="https://control.inorbit.ai"
 export INORBIT_API_URL="https://api.inorbit.ai"
 export INORBIT_API_KEY="foobar123"
 # Disable SSL for local development only
 export INORBIT_API_USE_SSL="true"
+# Optionally enable video streaming as camera "0"
+export INORBIT_VIDEO_URL=/dev/video0
```
[the remark this answers is not in the document]
Comment (miguelgarcia) [reply inorbit_edge/tests/demo/README.md:16]: Supports what you were expecting too. Anything that can be opened by OpenCV :D

File: inorbit_edge/video.py (hunk 1 above)
Comment (miguelgarcia) [reply inorbit_edge/video.py:30]: I had no idea about it, you rock Lean ! I'll do the change

Comment (miguelgarcia) [reply inorbit_edge/video.py]: I had no idea about that, super nice !!! fixed

File: inorbit_edge/tests/test_robot_session_callbacks.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def my_command_handler(command_name, args, options):
             call(topic="r/id_123/ros/loc/nav_goal"),
```
[the remark this answers is not in the document]
Comment (miguelgarcia) [reply inorbit_edge/tests/test_robot_session_callbacks.py:52]: fixed!

**Comment by miguelgarcia** [comment]
@leandropineda I addressed the feedback and also refactored the code a little. Let me know how it looks now!

**Review by leandropineda — APPROVED** [review]

**Pull request merged**